hazelcast/src/main/java/com/hazelcast/client/console/SqlConsole.java

```diff
@@ -224,6 +234,38 @@ private static void executeSqlCmd(
         }
     }
 
+    private static int printExplain(
+            SqlRowMetadata rowMetadata,
+            SqlResult sqlResult,
+            int[] colWidths,
+            Alignment[] alignments,
+            PrintWriter out
+    ) {
+        assert rowMetadata.getColumnCount() == 1 : "Explain query must produce only one column";
+        assert colWidths.length == 1 : "Explain query must produce only one column";
+
+        List<SqlRow> rows = new ArrayList<>(EXPLAIN_QUERY_OUTPUT_CAPACITY);
```

Comment:
[minor] If that code is single threaded (I believe it is), you can make ```rows``` a ```private static``` field of that class and create one instance of ```ArrayList``` to avoid litter. Just do ```.clear()``` in the end od that method.